src/Plugin/Block/BlogPrevNextBlock.php

```diff
@@ -0,0 +1,233 @@
+        $next_title = $next['title'];
+        $next_url = $next['url'];
+      }
+    }
+
+    return [
+      '#theme' => 'localgov_blogs_prev_next_block',
+      '#previous_url' => $previous_url,
+      '#previous_title' => $previous_title,
+      '#next_url' => $next_url,
+      '#next_title' => $next_title,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    // Get the created time of the current node.
+    $node = $this->routeMatch->getParameter('node');
+    if (!empty($node) && $node instanceof NodeInterface) {
+      // If there is node add its cachetag.
+      return Cache::mergeTags(parent::getCacheTags(), ['node:' . $node->id()]);
+    }
+    else {
+      // Return default tags instead.
+      return parent::getCacheTags();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['route']);
+  }
+
+  /**
+   * Lookup the previous node,youngest node which is still older than the node.
+   *
+   * @param string $node
+   *   Show current page node id.
+   *
+   * @return array
+   *   A render array for a previous node.
+   */
+  private function generatePrevious($node) {
+    return $this->generateNextPrevious($node, 'prev');
+  }
+
+  /**
+   * Lookup the next node,oldest node which is still younger than the node.
+   *
+   * @param string $node
+   *   Show current page node id.
+   *
+   * @return array
+   *   A render array for a next node.
+   */
+  private function generateNext($node) {
+    return $this->generateNextPrevious($node, 'next');
+  }
+
+  const DIRECTION__NEXT = 'next';
+
+  /**
+   * Lookup the next or previous node.
+   *
+   * @param string $node
+   *   Get current page node id.
+   * @param string $direction
+   *   Default value is "next" and other value come from
+   *   generatePrevious() and generatePrevious().
+   *
+   * @return array
+   *   Find the alias of the next node.
+   */
+  private function generateNextPrevious($node, $direction = self::DIRECTION__NEXT) {
+    $comparison_opperator = '>';
+    $sort = 'ASC';
+    $current_node_date = $node->get('localgov_blog_date')->value;
+    $current_langcode = $node->get('langcode')->value;
+    $current_blog_channel = $node->get('localgov_blog_channel')->target_id;
+
+    if ($direction === 'prev') {
+      $comparison_opperator = '<';
+      $sort = 'DESC';
+    }
+
+    // Lookup 1 node younger (or older) than the current node
+    // based upon the `localgov_blog_date` field.
+    $query = $this->entityTypeManager->getStorage('node');
+    $query_result = $query->getQuery();
+    $result = $query_result->condition('localgov_blog_date', $current_node_date, $comparison_opperator)
+      ->condition('type', 'localgov_blog_post')
+      ->condition('localgov_blog_channel', $current_blog_channel)
+      ->condition('status', 1)
+      ->condition('langcode', $current_langcode)
+      ->sort('localgov_blog_date', $sort)
```

Comment:
I made two posts on the same day. They don't get any previous next links. I think this is because the granularity of the date in day. If you have two posts on the same day it's not getting to them.

Reply:
@ekes I think this is becuase the blog date field is a date, not a date time. Simmilar issue in news as it's not got time granuallity. I suggest for this PR we add in checking for a page updated also, and then revise these date fields on blogs and news and convert them to datetime.

Reply:
@ekes @andybroomfield If eiether of you have some time to look at adding in a check for page updated that would be awesome - I have very limited time at the moment to look at this so any help would be greatly appreciated.